### PR TITLE
Suppress unsupported port name and missing containerPort with 1.3 sidecar

### PR DESCRIPTION
**Pull request by esnible**
Resolves https://github.com/istio/istio/issues/17017

### CODE REVIEW

File: istioctl/cmd/describe.go [hunk 1]
```diff
@@ -149,14 +150,17 @@ THIS COMMAND IS STILL UNDER ACTIVE DEVELOPMENT AND NOT READY FOR PRODUCTION USE.
 				return fmt.Errorf("can't parse sidecar config_dump: %v", err)
 			}
 
+			// If the sidecar is on Envoy 1.3 or higher, don't complain about empty K8s Svc Port name
+			istioVersion := getIstioVersion(&cd)
```
Comment (ayj) [inline istioctl/cmd/describe.go]: Can you explain why, i.e. port sniffing?

File: istioctl/cmd/describe.go [hunk 2]
```diff
@@ -236,7 +240,41 @@ func describe() *cobra.Command {
...
+}
+
+func containerPortOptional(istioVersion string) bool {
+	if istioVersion == noVersionMetadata {
+		return true
+	}
+	return supportsProtocolDetection(istioVersion)
+}
+
+func supportsProtocolDetection(istioVersion string) bool {
```
Comment (howardjohn) [inline istioctl/cmd/describe.go]: can you reuse functions like https://github.com/istio/istio/blob/2dc426df846f3c3f73393fac233e8785024787fb/pilot/pkg/networking/util/util.go#L270-L273 and https://github.com/istio/istio/blob/c9ba34bf2aa8a9129a7a5c18124d301bd6a1f2c6/pilot/pkg/model/context.go#L396-L419?

**Comment by esnible** [comment]
@ayj @howardjohn Now uses standard Pilot methods for parsing version string.  New tests added for cases of pre-1.3, 1.3, with and without container Port and node port name.

File: istioctl/cmd/describe_test.go [hunk 3]
```diff
@@ -260,6 +369,36 @@ DestinationRule: ratings for "ratings"
...
+			configs:          cannedIstioConfig,
+			k8sConfigs:       cannedK8sEnv,
+			args:             strings.Split("-n default experimental describe pod productpage-v1-7bbd79f8fd-k6j79", " "),
+			expectedOutput: `Pod: productpage-v1-7bbd79f8fd-k6j79
+   Pod Ports: 15090 (istio-proxy)
+--------------------
+Service: productpage
+   Port:  9080/UnsupportedProtocol
+   9080 is named "" which does not follow Istio conventions
+Authn: None
```
Comment (ayj) [inline istioctl/cmd/describe_test.go]: > `9080 is named ""`

Port 9080 is unnamed? 

Also, starting in 1.3, is it still convention to name ports if automatic port detection is available?

Comment (esnible) [reply istioctl/cmd/describe_test.go]: Do you think it is worthwhile to special-case ` named ""` as `unnamed`?  It isn't hard....

Lin told me that protocol detection was not enabled for inbound in 1.3.0.  If that is the case, the 1.3.0 client should not assume port detection is available.  When there is logic for it, I will replace the current `return false` with code specific to a version or feature flag.

Comment (ayj) [reply istioctl/cmd/describe_test.go]: Yes, probably worth it, particularly if its easy to add.

**Comment by esnible** [comment]
@ayj I have made the requested changes.  Can you re-review?

**Pull request merged**